templates/CRM/Contact/Form/Merge.tpl

```diff
@@ -419,7 +427,8 @@
     });
 
     // Call mergeBlock whenever a location type is changed
-    $('body').on('change', 'select[id$="locTypeId"],select[id$="typeTypeId"],input[id$="[operation]"],input[id$="[set_other_primary]"]', function(event){
+    $('select[id$="locTypeId"],select[id$="typeTypeId"],input[id$="[operation]"],input[id$="[set_other_primary]"]').on('change',
+      function(event){
```

Comment:
This bit doesn't work, can you revert it to my original? Or perhaps use the form.CRM_Contact_Form_Merge ?

The select elements are added dynamically, so the .on still needs to be assigned to the body, so it can pick anything up. Otherwise the 'add new' text doesn't get changed correctly.